server/api/apiController.js

```diff
@@ -87,12 +91,69 @@ module.exports = {
         text: params.text
       };
       emailUtils(mailOptions, function(error, results) {
-        if(error) {
+        if( error ) {
           console.error(error);
           res.status(500).end();
         }
         res.status(301).end('Ping sent');
       });
     });
+  },
+
+  postSignup: function(req, res, next) {
+    var params = req.body;
+    dbUtils.getUser(params, function(error, user) {
+      if( user ) {
+        res.end('Email already taken.');
+      } else {
+        authUtils.hashPassword(params.password, function(error, hash) {
+          if( error ) {
+            console.error(error);
+            res.status(500).end();
+          }
+          var userOptions = {
```

Comment:
I was thinking:
user.password_hash = hash;
and pass in user object to dbUtils.addUser (it parses out unwanted fields automagically)